[quickstart] Python RAG with FastAPI and LangGraph (#47)
auth0/docs-v2: auth4genai/snippets/get-started/langchain-fastapi-py/auth-for-rag.mdx

import { Prerequisites } from "/snippets/get-started/prerequisites/auth-for-rag.jsx";

<Prerequisites
  callbackUrl="http://localhost:8000/api/auth/callback"
  logoutUrl="http://localhost:5173"
/>

### Prepare the FastAPI app

**Recommended**: Use the starter template, clone the [Auth0 AI samples](https://github.com/auth0-samples/auth0-ai-samples) repository:

```bash wrap lines
git clone https://github.com/auth0-samples/auth0-ai-samples.git
cd auth0-ai-samples/authenticate-users/langchain-fastapi-py
```

The project is divided into two parts:

- `backend/` contains the backend code for the Web app and API written in Python using FastAPI and the LangGraph agent.
- `frontend/` contains the frontend code for the Web app written in React as a Vite SPA.

### Install dependencies

In the `backend` directory of your project, install the following dependencies:

- `auth0-ai-langchain`: [Auth0 AI SDK for LangChain](https://github.com/auth0-lab/auth0-ai-python/tree/main/packages/auth0-ai-langchain) built for GenAI applications powered by LangChain.

> - `langgraph`: For building stateful, multi-actor applications with LLMs.

Reviewer: ```suggestion
- `langgraph`: [LangGraph](https://pypi.org/project/langgraph/) for building stateful, multi-factor applications with LLMs.
```

Author: it is multi-actor as in multi-agent

> - `langgraph`: For building stateful, multi-actor applications with LLMs.

Author: ```suggestion
- `langgraph`: [LangGraph](https://pypi.org/project/langgraph/) for building stateful, multi-actor applications with LLMs.
```

- `langchain-openai`: [OpenAI](https://python.langchain.com/docs/integrations/chat/openai) provider for LangChain.
- `langgraph-cli`: LangGraph CLI for running a local LangGraph server.
- `openfga-sdk`: OpenFGA SDK for Fine-Grained Authorization

Make sure you have [uv](https://docs.astral.sh/uv/) installed and run the following command to install the dependencies:

```bash wrap lines
cd backend
uv sync
uv add "auth0-ai-langchain>=1.0.0b3" openfga-sdk langgraph langchain-openai "langgraph-cli[inmem]" --prerelease=allow
```

### Update the environment file

> Copy the `.env.example` file to `.env` and update the variables with your Auth0 credentials. You can find your Auth0 domain, client ID and client secret in the application you created in the Auth0 dashboard.

Reviewer: ```suggestion
Copy the `.env.example` file to `.env` and update the variables with your Auth0 credentials. You can find your Auth0 domain, client ID, and client secret in the application you created in the Auth0 Dashboard.
```
